_posts/2013-04-06-developing-addons-and-blueprints.md

```diff
@@ -594,3 +594,68 @@ running `ember init` in your project root.
 For a good walkthrough of the (recent) development of a real world
 addon, take a look at:
 [Creating a DatePicker Ember CLI addon](http://edgycircle.com/blog/2014-creating-a-datepicker-ember-addon)
+
+### In-Repo Addons
+
+Addons sepecific to your project can be created inside your repo and are
+generated in the projects `lib` directory in a folder with the name of
+the in-repo addon, e.g. `/lib/in-repo-addon-name` and follow the same
+file structure as a normal *addon*.
+
+Some advantages of using an in-repo-addon, instead of an addon outside of
+your application (repo), are:
+
+- Sandbox for developing a feature that you may want to share as an
+  addon in the future
+- Having all the benefits of addon isolation but without the need to
+  publish or `npm link`
+
+Use `in-repo-addon` argument with the `ember generate` command:
+
+{% highlight bash %}
+ember generate in-repo-addon in-repo-addon-name
+{% endhighlight %}
+
+(Replace `in-repo-addon-name` with the name of your addon.)
+
+Shorthand arguments `-ir` and `-in-repo` can be used in place of `--in-repo-addon`.
+
+For your in-repo addon stylesheet, name the file `addon.css` and place
+it in the styles directory, e.g `/lib/in-repo-addon-name/addon/styles/addon.css`
+This avoids any conflict with the parent application's `app.css` file
+
+In order to complile HTMLBars templates that are part of your in-repo addon,
+your `package.json` file will need to include following dependencies:
+
+- `babel-plugin-htmlbars-inline-precompile`
+- `ember-cli-babel`
+- `ember-cli-htmlbars`
+- `ember-cli-htmlbars-inline-precompile`
+
+(Use the same versions found in your Ember CLI Application's `package.json`)
+
+To ensure the you can use babel.js and related polyfills with your in-repo
+addon add babel options to the `included` hook of the in-repo index.js file:
+
+{% highlight javascript %}
+module.exports = {
+  name: 'in-repo-addon-name',
+
+  isDevelopingAddon: function() {
+    return true;
+  },
+
+  included: function(app, parentAddon) {
+    var target = (parentAddon || app);
+    target.options = target.options || {};
+    target.options.babel = target.options.babel || { includePolyfill: true };
+    return this._super.included(target);
+  }
+};
+{% endhighlight %}
+
+To create a blueprint for your in-repo-addon:
```

Comment:
I just re-read this and realize now that I misunderstood the actual intent was to generate a blueprint inside an in-repo-addon. It might need some editing to be a little more clear.